framework/include/utils/MathFVUtils.h

```diff
@@ -98,12 +110,38 @@ typename libMesh::CompareTypes<T, T2>::supertype
 linearInterpolation(const T & value1,
                     const T2 & value2,
                     const FaceInfo & fi,
-                    const bool one_is_elem)
+                    const bool one_is_elem,
+                    const InterpMethod interp_method = InterpMethod::Average)
 {
-  const auto coeffs = interpCoeffs(InterpMethod::Average, fi, one_is_elem);
+  mooseAssert(interp_method == InterpMethod::Average ||
+                  interp_method == InterpMethod::SkewCorrectedAverage,
+              "The selected interpolation function only works with average or skewness-corrected "
+              "average options!");
+  const auto coeffs = interpCoeffs(interp_method, fi, one_is_elem);
   return coeffs.first * value1 + coeffs.second * value2;
 }
 
+/**
+ * Linear interpolation with skewness correction using the face gradient.
+ * See more info in Moukalled Chapter 9. The correction involves a first order
+ * Taylor expansion around the intersection of the cell face and the line
+ * connecting the two cell centers.
+ */
+template <typename T, typename T2, typename T3>
+typename libMesh::CompareTypes<T, T2>::supertype
+skewCorrectedlinearInterpolation(const T & value1,
+                                 const T2 & value2,
+                                 const T3 & face_gradient,
+                                 const FaceInfo & fi,
+                                 const bool one_is_elem)
+{
+  const auto coeffs = interpCoeffs(InterpMethod::SkewCorrectedAverage, fi, one_is_elem);
+
+  auto value = (coeffs.first * value1 + coeffs.second * value2) +
+               face_gradient * (fi.faceCentroid() - fi.rIntersection());
```

Comment:
Actually, I looked at Moukalled and I am having issues with some of his explanation, not only on this topic but in general. I looked at OF too and they average the gradients to the face as well. However, they have other issue with the skewness-correction. I researched the topic beyond these two to check the expected convergence rates or the mainsteam approaches in the literature. At the end I added the regular gradient instead of the uncorrected gradient because it yielded the lowest errors. We can definitely save a few operation here, if needed. Also, I have never seen the dFf/(dCf+dFf) weights for unstructured meshes. Therefore, I added a new weighting factor gc_skewed in faceinfo that uses the dFf'/dFC which is the standard in OF and other FV codes for every mesh. For structured quads it is the same. I think Moukalled is inconsistent and vague in some cases. I like the way he describes the general concepts but he doesn't give convergence estimates/proofs and his OF examples are from 10 years ago. I can spend some time tomorrow to properly compare the two approaches if you think it is necessary. I'll add some more comments in the header.

Reply:
I have also had similar frustrations with Moukalled over time.

This PR is already big enough that we don't need to add a bunch of exodiffs and regolding on top of it, but I'm all for making the `gC` computation consistent with other codes

Reply:
That could go in a follow-on PR

Reply:
I"ve opened https://github.com/idaholab/moose/issues/19361 for this

Reply:
I ended up adding a comment in the source file of MooseVariableFV instead of the MathFVUtils. I think the interpolation itself follows what is introduced in Moukalled, only the face gradient is different which is now (next commit) mentioned in the source file right before we compute the corresponding face_gradient.